frontend/src/console/components/InstanceSettings/InstanceSettings.tsx

```diff
@@ -31,6 +34,19 @@ export class InstanceSettingsComponent extends React.PureComponent<
     };
   }
 
+  public componentDidUpdate(prevProps: Props) {
+    // Fill fields after finishing loading data
+    if (
+      prevProps.activeInstance.data === null &&
+      this.props.activeInstance.data
+    ) {
+      this.setState({
+        instanceName: this.props.activeInstance.data.instanceName,
```

Comment:
Whenever you have similar data in `State` and in `Props`, where the state is the possibly new value that the user is entering via a form, and the props is the existing value, I recommend naming the state variable something like `newInstanceName`. Then it's much less confusing to know why it sometimes uses `this.props. ... .instanceName` and sometimes `this.state.instanceName`